src/group/mls_group/test_mls_group.rs

```diff
@@ -125,3 +127,183 @@ fn test_failed_groupinfo_decryption() {
+                &[],
+                false,
+            )
+            .expect("Error creating commit");
+        let commit = match mls_plaintext_commit.content() {
+            MLSPlaintextContentType::Commit((commit, _)) => commit,
+            _ => panic!("Wrong content type"),
+        };
+        assert!(!commit.has_path() && kpb_option.is_none());
+        // Check that the function returned a Welcome message
+        assert!(welcome_bundle_alice_bob_option.is_some());
+
+        group_alice
+            .apply_commit(&mls_plaintext_commit, epoch_proposals, &[])
+            .expect("error applying commit");
+        let ratchet_tree = group_alice.tree().public_key_tree_copy();
+
+        let group_bob = MlsGroup::new_from_welcome(
+            welcome_bundle_alice_bob_option.unwrap(),
+            Some(ratchet_tree),
+            bob_key_package_bundle,
+        )
+        .unwrap();
+
+        // Make sure that both groups have the same public tree
+        if group_alice.tree().public_key_tree() != group_bob.tree().public_key_tree() {
+            _print_tree(&group_alice.tree(), "Alice added Bob");
+            panic!("Different public trees");
+        }
+
+        // === Bob updates and commits ===
+        let bob_update_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &bob_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+
+        let update_proposal_bob = group_bob.create_update_proposal(
+            &[],
+            &bob_credential_bundle,
+            bob_update_key_package_bundle.key_package().clone(),
+        );
+        let (mls_plaintext_commit, _welcome_option, _kpb_option) = group_bob
+            .create_commit(
+                &[],
+                &bob_credential_bundle,
+                &[&update_proposal_bob],
+                &[],
+                false, /* force self update */
+            )
+            .unwrap();
+
+        // Now we break Alice's HPKE ciphertext in Bob's commit by breaking
+        // apart the commit, manipulating the ciphertexts and the piecing it
+        // back together.
+        let (commit, confirmation_tag) = match &mls_plaintext_commit.content {
+            MLSPlaintextContentType::Commit((commit, confirmation_tag)) => {
+                (commit, confirmation_tag)
+            }
+            _ => panic!("Bob created a commit, which does not contain an actual commit."),
+        };
+
+        let commit = commit.clone();
+
+        let path = commit.path.unwrap();
+
+        // For simplicity, let's just break all the ciphertexts.
+        let mut new_nodes = Vec::new();
+        for node in path.nodes {
+            let mut new_eps = Vec::new();
+            for c in node.encrypted_path_secret {
+                let mut c_encoded = c.encode_detached().unwrap();
+                let c_last_bits = c_encoded.pop().unwrap();
+                c_encoded.push(c_last_bits.reverse_bits());
```

Comment:
I've seen this code snippet before. It should probably go into a helper function.

Reply:
Done